src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/NativeFormatEnumInfoImplementation.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using global::System;
+using global::System.Reflection;
+using global::System.Collections.Generic;
+
+using global::Internal.Runtime.Augments;
+
+using global::Internal.Reflection.Core;
+using global::Internal.Reflection.Core.Execution;
+
+using global::Internal.Metadata.NativeFormat;
+
+namespace Internal.Reflection.Execution
+{
+    internal sealed class NativeFormatEnumInfoImplementation : EnumInfoImplementation
+    {
+        public NativeFormatEnumInfoImplementation(Type enumType, MetadataReader reader, TypeDefinitionHandle typeDefinitionHandle) : base(enumType)
+        {
+            _reader = reader;
+            _typeDefinition = typeDefinitionHandle.GetTypeDefinition(reader);
+        }
+
+        protected override KeyValuePair<String, ulong>[] ReadNamesAndValues()
```

Comment:
Prefer "protected sealed override KeyValuePair.." - in keeping with the Reflection code style (and less manual work for the human reader to know that this is the final override...)